### DISCUSSION TITLE
How to use v6 control with custom change handler

### DISCUSSION THREAD

**Discussion by nlrowe** [react-hook-form/react-hook-form#2293] [Q&A] [answered]
With v5, I have the following `Select` wrapped in a `Controller`:
```
<Controller
    name={'id'}
    as={Select}
    options={options}
    control={control}
    rules={{validate: () => validateSelection(index)}}
    onChange={handleChange.bind(this, append, index)}
/>
```
`react-hook-form` takes my custom change handler and calls it when it needs to, everything works fine.

After migrating to v6, my change handler no longer gets called. I went to the v6 `Controller` documentation and found the following rules for using the `as` prop:
- `onChange`, `onBlur`, and `value` props are injected into the component referenced in the `as` prop
- all other props will be forwarded to the `as` component

So I assume my custom change handler is now being ignored as `react-hook-form` is injecting its own internal handler into the component.

I looked at the documentation again and found that there is a new prop called `render`, which is noted to 'Simplify integrating with external controlled components with non-standard prop names.' My `Select` component does not have non-standard prop names, but I think maybe this is the intended way to inject a custom change handler. So I change my code to the following:
```
<Controller
    name={'id'}
    render={({value, onChange, onBlur}) => (
        <Select
            value={value}
            options={options}
            onChange={handleChange.bind(this, append, index)}
            onBlur={onBlur}
        />
    )}
    control={control}
    rules={{validate: () => validateSelection(index)}}
/>
```
After this change the custom change handler is called when the select is changed and everything works fine. However, I now have this `onChange` function that the render function is trying to pass down to me that I never used.

Two questions:
1. Is it intended that you use the `render` prop in order to use a custom change handler for the rendered component?
2. If I am supposed to use the `render` prop, do I ignore the `onChange` function that it is trying to pass me? Or do I need to call that function in my change handler?

**Comment by bluebill1049** [comment, marked as answer]
can you do this:
```
render(({ onChange }) => {
  return <input onChange={(e) => {  onChange(e.target.value); myMethod(); }} />
})
```

**Reply by nlrowe** [reply]
Yes of course. Is that the intended way to handle the scenario of using a custom change handler with a `Controller` component?

**Reply by bluebill1049** [reply]
yes.

**Reply by bluebill1049** [reply]
basically controller onChange is retrieving input value.
